packages/components/organisms/f-checkout/src/store/checkout.module.js

```diff
@@ -372,9 +372,11 @@ export default {
             }
 
             if (!addressCoords && state.authToken) {
-                const { data } = await addressGeocodingApi.getGeoLocation(url, postData, timeout, state);
+                const response = await addressGeocodingApi.getGeoLocation(url, postData, timeout, state);
```

Comment:
You could still destructure here and use `data` in the check below?

Reply:
The error we're getting is 
```
Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
``` 

so just trying to avoid that and confirm that it's this line causing it.

Reply:
And it's failing on this line? I'd expect to see that error if you were restructuring using square brackets ().

Looks to me like [it'd be the mutation throwing that error](https://github.com/justeat/fozzie-components/blob/b313ca208377eb4da9498183cc6cb1a0a3e93435/packages/components/organisms/f-checkout/src/store/checkout.module.js#L566) as it's attempting to destrucure the lat/lng values — if you pass an invalid value it'll throw.

In any case, adding the `if` guard here should fix it, you could put the `data` restructure back and just do the check against `data?.geometry`?

Reply:
You could also move the guard into the mutation itself and only destructure the lat/lngs if the payload has a value. This could be a better approach as the mutation is called in a couple of places.

Reply:
One more thing, you can probably write a unit test for the mutation that will throw that non-iterable error in its current state.

Reply:
Thanks @DamianMullins - for some reason I didn't even consider the mutation. I will update the PR now.